test/internal/cmd/io.go

```diff
@@ -38,12 +41,45 @@ func (b *BufferedIO) Output() (_ string, err error) {
 func (b *BufferedIO) TestOutput(tb testing.TB, out string, err error) {
 	tb.Helper()
 
-	outGive, errGive := b.Output()
-	if !errors.Is(errGive, err) {
-		tb.Fatalf("got stderr: %v; wanted: %v", errGive, err)
+	outGot, errGot := b.Output()
+	if !errors.Is(errGot, err) {
+		tb.Fatalf("got stderr: %v; wanted: %v", errGot, err)
+	}
+
+	out = strings.ToLower(strings.TrimSpace(out))
+	outGot = strings.ToLower(strings.TrimSpace(outGot))
+	if diff := cmp.Diff(out, outGot); diff != "" {
+		tb.Fatalf("stdout mismatch (-want +got):\n%s", diff)
```

Comment:
`stdout` could be huge, probably should avoid logging the entire diff. Same for the function below.

Reply:
i tried that, but i found that whenever the test failed, having the stdout in the logs was pretty helpful to debug the failure immediately. 
otherwise i went down a rabbit hole of replicating locally and then logging stdout anyways

Reply:
We can limit the size of the diff that we output. For example, if it is longer than 50 chars we only print first 50.

Reply:
can i defer that to a different PR? that would likely complicate this code unnecessarily